presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -376,10 +376,15 @@ public void requestSystemMemoryRevoking()
 
     public void resetSystemMemoryRevokingRequested()
     {
-        // It seems that we can blindly replace the future here.
-        // The only negative outcome of race condition here would be that we will loose memory
-        // revoking request. Another one will happen then anyway.
-        systemMemoryRevokingRequestedFuture.set(SettableFuture.create());
+        while (true) {
+            SettableFuture<?> currentFuture = systemMemoryRevokingRequestedFuture.get();
```

Comment:
not really.
The future is only changed in this method. So it can loop only in case when two of such methods are run simultanously. And then the one which looped will exit on first if on next iteration. As the winner of race replaced the future with whine which is not `done()`. 

Actually as this is only method which changes value of this future I will change it to:

```java
            SettableFuture<?> currentFuture = systemMemoryRevokingRequestedFuture.get();
            if (!currentFuture.isDone()) {
                return;
            }
            systemMemoryRevokingRequestedFuture.compareAndSet(currentFuture, SettableFuture.create())
        }
```

It will have very similar behaviour and not have scary loop.